runtime/runtime.go

```diff
@@ -34,14 +34,10 @@ func init() {
 	failpoints = make(map[string]*Failpoint)
 	envTerms = make(map[string]string)
 	if s := os.Getenv("GOFAIL_FAILPOINTS"); len(s) > 0 {
-		// format is <FAILPOINT>=<TERMS>[;<FAILPOINT>=<TERMS>;...]
-		for _, fp := range strings.Split(s, ";") {
-			fpTerm := strings.Split(fp, "=")
-			if len(fpTerm) != 2 {
-				fmt.Printf("bad failpoint %q\n", fp)
-				os.Exit(1)
-			}
-			envTerms[fpTerm[0]] = fpTerm[1]
+		if fpMap, err := parseFailpoints(s); err != nil {
+			panic(err)
```

Comment:
I think the previous exit behavior was more understandable for user. There issue is with user provided environment environment, but panic implies issue was with code. 
```
				fmt.Printf("bad failpoint %q\n", fp)
				os.Exit(1)
```

Reply:
good point & updated.